### PR TITLE
SVG1.1 requires <use> href attr to be qualified as xlink:href

### PR DESCRIPTION
**Pull request by anthrotype**
We were using the unqualified `href` attribute notation with the `<use>` element, which is [only supported in SVG2](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href). For `<use>` to work properly on Safari and Adobe apps we need to declare the `xlink` namespace and qualify the attribute as `xlink:href`

Fixes https://github.com/googlefonts/nanoemoji/issues/264 and https://github.com/googlefonts/nanoemoji/issues/266

Supersedes https://github.com/googlefonts/picosvg/pull/207 and https://github.com/googlefonts/nanoemoji/pull/270

Many thanks @yisibl for helping debugging and fixing this!

### CODE REVIEW

**Comment by yisibl** [comment]
Let me add some explanations so that everyone can know what has been changed here.

The `<use>` element must use the `xlink:href` attribute, not `href`. Otherwise it will not take effect in Safari/Mac/iOS.

Although the SVG2 specification recommends using href instead of `xlink:href`, the svg-native specification says:

> deprecated xlink attributes except for 'xlink:href'
> https://svgwg.org/specs/svg-native/#commonattributes

SVG in OT fonts should follow the svg-native specification.

File: tests/clocks_rects_picosvg.ttx [hunk 1]
```diff
@@ -64,7 +64,7 @@
...
...#424242" id="glyph2::3"/><path d="M67,64 A3 3 0 1 1 61,64 A3 3 0 1 1 67,64 Z" fill="white" id="glyph2::4"/><path d="M19.89,64.13 A4 4 0 1 1 11.89,64.13 A4 4 0 1 1 19.89,64.13 Z" fill="#757575" id="glyph2::5"/><use xlink:href="#glyph2::5" x="48" y="-48"/><use xlink:href="#glyph2::5" x="48" y="48"/><use xlink:href="#glyph2::5" x="96"/><path d="M64,8 C94.88,8 120,33.12 120,64 S94.88,120 64,120 S8,94.88 8,64 S33.12,8 64,8 M64,4 C30.86,4 4,30.86 4,64 S30.86,124 64,124 S124,97.14 124,64 S97.14,4 64,4 L64,4 Z" fill="#424242" opacity="0.2" id="glyph2::9"/></g><g id="glyph3" transform="matrix(0.781 0 0 0.781 0 -100)"><use xlink:href="#glyph2::0"/><path d="M63.8,63.98 H64.2 C66.3,63.98 68,62.28 68,60.18 V27.78 C68,25.68 66.3,23.98 64.2,23.98 H63.8 C61.7,23.98 60,25.68 60,27.78 V60.18 C60,62.28 61.7,63.98 63.8,63.98 Z" fill="purple"/><path d="M63.78,63.81 L63.97,64.16 C64.98,66 67.29,66.68 69.13,65.67 L88.07,55.31 C89.91,54.3 90.59,51.99 89.58,50.15 L89.39,49.8 C88.38,47.96 86.07,47.28 84.23,48.29 L65.29,58.65 C63.45,59.66 62.77,61.97 63.78,63.81 Z" fill="red"/><use xlink:href="#glyph2::3"/><use xlink:href="#glyph2::4"/><use xlink:href="#glyph2::5"/><use xlink:href="#glyph2::9"/></g></svg>]]>
```
Comment (yisibl) [inline tests/clocks_rects_picosvg.ttx:67]: It is best to add a more concise test case, which is a good choice:

```svg
<svg width="200" height="200" viewBox="0 0 200 200" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <circle fill="#F49924" cx="100" cy="100" r="100" />
    <ellipse fill="#7E4418" cx="59.545" cy="71.818" rx="15" ry="21.818" />
    <ellipse fill="#7E4418" cx="140.455" cy="71.818" rx="15" ry="21.818" />
</svg>
```

**Review by yisibl — COMMENTED** [review]
LGTM

File: tests/clocks_rects_picosvg.ttx (hunk 1 above)
Comment (anthrotype) [reply tests/clocks_rects_picosvg.ttx:67]: ok, I'll add that.

File: tests/reused_shape_2_picosvg.ttx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    <TTGlyph name=".space"/><!-- contains no outline data -->
+
+    <TTGlyph name="e000"/><!-- contains no outline data -->
+
+  </glyf>
+
+  <SVG>
+
+    <svgDoc endGlyphID="2" startGlyphID="2">
+      <![CDATA[<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1"><defs/><g id="glyph2" transform="matrix(0.5 0 0 0.5 0 -100)"><path d="M200,100 A100 100 0 1 1 0,100 A100 100 0 1 1 200,100 Z" fill="#F49924"/><path d="M74.545,71.818 A15 21.818 0 1 1 44.545,71.818 A15 21.818 0 1 1 74.545,71.818 Z" fill="#7E4418" id="glyph2::1"/><use xlink:href="#glyph2::1" x="80.91"/></g></svg>]]>
```
Comment (yisibl) [inline tests/reused_shape_2_picosvg.ttx:49]: In the SVG in OT font, can <defs/> be deleted?

Comment (anthrotype) [reply tests/reused_shape_2_picosvg.ttx:49]: what do you mean?

Comment (yisibl) [reply tests/reused_shape_2_picosvg.ttx:49]: If you delete the empty `<defs/>`, I think the file size will be smaller.

Comment (anthrotype) [reply tests/reused_shape_2_picosvg.ttx:49]: 7 bytes.. but that's a separate issue

Comment (yisibl) [reply tests/reused_shape_2_picosvg.ttx:49]: yeah.

**Pull request merged**